drivers/hmis/app/graphql/mutations/bulk_review_external_submissions.rb

```diff
@@ -22,36 +22,39 @@ def resolve(external_submission_ids:)
       access_denied! unless current_permission?(permission: :can_manage_external_form_submissions, entity: project)
       access_denied! if definitions.any?(&:updates_client_or_enrollment?) && !current_permission?(permission: :can_edit_enrollments, entity: project)
 
-      already_reviewed = submissions.filter { |s| s.enrollment_id.present? || s.status == 'reviewed' }
-      error_out('Submissions are already processed: ' + already_reviewed.pluck(:id).inspect) if already_reviewed.any?
-
       should_auto_enter = project.should_auto_enter?
 
-      HmisExternalApis::ExternalForms::FormSubmission.transaction do
-        submissions.each do |record|
-          record.status = 'reviewed'
-          record.run_form_processor(current_user, project: project)
+      # maps record IDs to error messages
+      failed_to_review = {}
+
+      submissions.each do |record|
+        raise ArgumentError, 'Already reviewed' if record.enrollment_id.present? || record.status == 'reviewed'
 
-          if record.enrollment&.new_record?
-            error_out(record.enrollment.client.errors.full_messages) unless record.enrollment.client.valid?
-            error_out(record.enrollment.errors.full_messages) unless record.enrollment.valid?
+        record.status = 'reviewed'
+        record.run_form_processor(current_user, project: project)
 
-            record.enrollment.client.save!
-            should_auto_enter ? record.enrollment.save_and_auto_enter! : record.enrollment.save_in_progress!
-          end
+        if record.enrollment&.new_record?
+          raise ArgumentError, record.enrollment.client.errors.full_messages unless record.enrollment.client.valid?
+          raise ArgumentError, record.enrollment.errors.full_messages unless record.enrollment.valid?
 
-          record.form_processor.save!
-          record.save!
+          record.enrollment.client.save!
+          should_auto_enter ? record.enrollment.save_and_auto_enter! : record.enrollment.save_in_progress!
         end
-      end
 
-      { success: true }
-    end
+        record.form_processor.save!
+        record.save!
+      rescue StandardError => e
+        failed_to_review[record.id] = e.message
+      end
 
-    protected
+      if failed_to_review.any?
+        base_message = "Bulk review failed on #{failed_to_review.size} of #{submissions.size} records."
+        error_message = base_message + "\n" + failed_to_review.map { |id, message| "\tSubmission #{id}: #{message}" }.join("\n")
+        display_message = base_message + ' This may indicate that the following submissions are spam, or have already been reviewed: ' + failed_to_review.keys.join(', ')
+        raise HmisErrors::ApiError.new(error_message, display_message: display_message)
```

Comment:
The message looks good to me!

Reply:
Since we're raising instead of returning a ValidationError, we will get notified in Sentry. I think that's probably the right call since it _could_ point to a problem with our implementation.

Reply:
Yes, this was my thought too. Also because we're curious to see how much spam we will actually get, if any. If it gets really noisy for reasons we can't control then we can turn off the sentry alert by returning a `ValidationError` instead.